orca-retrofit/src/main/groovy/com/netflix/spinnaker/orca/retrofit/exceptions/BaseRetrofitExceptionHandler.groovy

```diff
@@ -36,6 +36,23 @@ abstract class BaseRetrofitExceptionHandler implements ExceptionHandler {
     return isIdempotentRequest(e) && (isNetworkError(kind) || isGatewayErrorCode(kind, responseCode) || isThrottle(kind, responseCode))
   }
 
+  boolean shouldRetry(Exception e, String kind, String httpMethod, Integer responseCode) {
```

Comment:
I suspect there's a way to reduce some duplication here by keeping only one shouldRetry method with the additional `httpMethod` argument (or maybe keeping the existing one without and calling the new one with `httpMethod` as null).  Then, if `httpMethod` is null, call `findHttpMethodAnnotation` to populate it, and then we end up with only one `isIdempotentRequest` method as well.

Reply:
I think keeping the existing one makes sense for now but I can definitely reduce some duplication inside the two methods

Reply:
LGTM, thanks!